### ISSUE TITLE
[clangd] IWYU pragma only works on top-level includes

### ISSUE THREAD

**Issue by yshui** [llvm/llvm-project#98852]
This works:

```c++
#include "unused.h" // IWYU pragma: keep
```

But this doesn't:

```c++
namespace N {
#include "unused.h" // IWYU pragma: keep
}
```

**Comment by kadircet** [comment]
what's the version of clangd? it should be fixed with https://github.com/llvm/llvm-project/commit/7837110ed8efdd510516c849178a7af28b93aea4

**Comment by HighCommander4** [comment]
> what's the version of clangd? it should be fixed with [7837110](https://github.com/llvm/llvm-project/commit/7837110ed8efdd510516c849178a7af28b93aea4)

(Which is present in clangd 18 and later.)

**Comment by yshui** [comment]
I can confirm it is fixed in 18! I was using 17.0.6
